### ISSUE TITLE
[Feature Request] Cyclical Learning Rates

### ISSUE THREAD

**Issue by Kaixhin** [pytorch/pytorch#1909]
[Cyclical learning rates](https://arxiv.org/abs/1506.01186) seem to be becoming more popular. Given that we now have a [lr scheduler](https://github.com/pytorch/pytorch/pull/1370), it would be nice to have this available as another schedule scheme. Pinging @ajbrock, who has experience with this (unlike me haha).

**Comment by ajbrock** [comment]
I'm down, though I wasn't aware of the LR scheduler API so I'll need to dig into that and learn it before I can contribute. I personally have seen more uses of [SGDR](https://arxiv.org/abs/1608.03983) style learning rates than cyclic learning rates (the only two cites on the CLR paper are self-cites) but I don't think either should be too tough to throw together.

**Comment by Jiaming-Liu** [comment]
Hi. I am the puller of LR Scheduler. IMO, you should be able to easily implement it using LambdaLR class. @ajbrock
